### PR TITLE
Two parts

### PR DESCRIPTION
**Pull request by SteveHNH**
Support for the second part

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -319,6 +320,7 @@ def get(self):
         values['hash'] = payload_id  # provided for backward compatibility
         values['size'] = size
         values['service'] = service
+        values['metadata'] = json.loads(self.metadata)
```
Comment (kylape) [inline app.py]: It looks like `metadata` could be `None`

File: app.py [hunk 2]
```diff
@@ -382,13 +389,14 @@ def write_data(self, body):
             return
         else:
             tracking_id = str(self.request.headers.get('Tracking-ID', "null"))
-            service = split_content(self.request.files['upload'][0]['content_type'])
+            self.metadata = self.arguments.get('metadata')[0].decode('utf-8') if self.arguments.get('metadata') else None
```
Comment (kylape) [inline app.py]: I thought the metadata was a separate part in the body.  wouldn't that be accessed via the `files` attribute?

Comment (SteveHNH) [reply app.py]: Tornado had issues with two form parts. It was only picking up the actual file upload. I had to change around the way that worked completely. I tested this with `-F 'upload=@file' -F 'metatdata={"some-json": "stuff"}` and it works using this method. Using the old method, the only thing that got picked up was the upload field

File: app.py (hunk 1 above)
Comment (SteveHNH) [reply app.py]: It could be, but is that bad?

Comment (SteveHNH) [reply app.py]: nevermind. I see waht you're saying

**Pull request merged**